airbyte-webapp/src/services/connectorBuilder/ConnectorBuilderStateService.tsx

```diff
@@ -148,7 +125,68 @@ export const ConnectorBuilderStateProvider: React.FC<React.PropsWithChildren<unk
   return <ConnectorBuilderStateContext.Provider value={ctx}>{children}</ConnectorBuilderStateContext.Provider>;
 };
 
-export const useConnectorBuilderState = (): Context => {
+export const ConnectorBuilderAPIProvider: React.FC<React.PropsWithChildren<unknown>> = ({ children }) => {
```

Comment:
I like this split of the contexts, I think it makes sense conceptually  👍 

Only thing I'm not sure of is the naming - maybe we could call them something like `ConnectorBuilderFormState` and `ConnectorBuilderTestState`? I want to make sure this doesn't get confused with https://github.com/airbytehq/airbyte/blob/master/airbyte-webapp/src/services/connectorBuilder/ConnectorBuilderApiService.ts